src/core/assertions.js

```diff
@@ -92,36 +98,43 @@ exports.assertObjectMatchSpec = (object, spec, exact = false) => {
         const currentValue = _.get(object, field)
         const expectedValue = Cast.value(value)
 
-        switch (matcher) {
-            case 'match':
-            case 'matches':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
-                ).to.match(new RegExp(expectedValue))
-                break
+        if (matchRegex.exec(matcher) !== null) {
+            expect(
+                currentValue,
+                `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
+            ).to.match(new RegExp(expectedValue))
+        }
 
-            case 'contain':
-            case 'contains':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not contain '${expectedValue}'`
-                ).to.contain(expectedValue)
-                break
+        if (containRegex.exec(matcher) !== null) {
+            expect(
+                currentValue,
+                `Property '${field}' (${currentValue}) does not contain '${expectedValue}'`
+            ).to.contain(expectedValue)
+        }
 
-            case 'defined':
-            case 'present':
-                expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
-                break
+        if (definedRegex.exec(matcher) !== null) {
+            expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
+        }
 
-            case 'equal':
-            case 'equals':
-            default:
-                expect(
-                    currentValue,
-                    `Expected property '${field}' to equal '${value}', but found '${currentValue}'`
-                ).to.be.deep.equal(expectedValue)
+        const dateOffsetRegexMatch = dateOffsetRegex.exec(matcher)
+        const expectedValueRegexMatch = dateValueRegex.exec(expectedValue)
+        if (dateOffsetRegexMatch !== null && expectedValueRegexMatch !== null) {
+            const [, format, locale] = dateOffsetRegexMatch
+            const [, amount, , unit] = expectedValueRegexMatch
+            const expectedDate = moment()
+                .add(amount, unit)
+                .locale(locale)
+                .format(format)
+            expect(
+                currentValue,
+                `Expected property '${field}' to equal '${expectedDate}', but found '${currentValue}'`
+            ).to.be.deep.equal(expectedDate)
         }
+
+        expect(
```

Comment:
I tend to prefer to ensure we're *matching a matcher*, otherwise I'm afraid that users could miss a mismatch

Reply:
Agreed. I put it back

Reply:
Have you pushed the change?

Reply:
yep